src/parser/classish_statement.rs

```diff
@@ -0,0 +1,341 @@
+        }
+
+        if self.current.kind == TokenKind::Function {
+            return self.method(ClassishDefinitionType::Enum, vec![]);
+        }
+
+        let member_flags = self.enum_members_flags()?;
+
+        match &self.current.kind {
+            TokenKind::Const => self.parse_classish_const(member_flags),
+            TokenKind::Function => self.method(
+                ClassishDefinitionType::Enum,
+                member_flags.iter().map(|t| t.clone().into()).collect(),
+            ),
+            _ => expected_token_err!(["`const`", "`function`"], self),
+        }
+    }
+
+    fn complete_class_statement(
+        &mut self,
+        class_type: ClassishDefinitionType,
+    ) -> ParseResult<Statement> {
+        if self.current.kind == TokenKind::Use {
+            return self.parse_classish_uses();
+        }
+
+        if self.current.kind == TokenKind::Var {
+            return self.parse_classish_var();
+        }
+
+        if self.current.kind == TokenKind::Const {
+            return self.parse_classish_const(vec![]);
+        }
+
+        if self.current.kind == TokenKind::Function {
+            return self.method(class_type, vec![]);
+        }
+
+        let member_flags = self.class_members_flags()?;
+
+        match &self.current.kind {
+            TokenKind::Const => self.parse_classish_const(member_flags),
+            TokenKind::Function => self.method(
+                class_type,
+                member_flags.iter().map(|t| t.clone().into()).collect(),
+            ),
+            // TODO
+            TokenKind::Variable(_) => {
+                let var = self.var()?;
+                let mut value = None;
+
+                if self.current.kind == TokenKind::Equals {
+                    self.next();
+                    value = Some(self.expression(Precedence::Lowest)?);
+                }
+
+                self.semi()?;
+
+                Ok(Statement::Property {
+                    var,
+                    value,
+                    r#type: None,
+                    flags: member_flags.into_iter().map(|f| f.into()).collect(),
+                })
+            }
+            TokenKind::Question
+            | TokenKind::Identifier(_)
+            | TokenKind::QualifiedIdentifier(_)
+            | TokenKind::FullyQualifiedIdentifier(_)
+            | TokenKind::Array
+            | TokenKind::Null => {
+                let prop_type = self.type_string()?;
+                let var = self.var()?;
+                let mut value = None;
+
+                if self.current.kind == TokenKind::Equals {
+                    self.next();
+                    value = Some(self.expression(Precedence::Lowest)?);
+                }
+
+                // TODO: Support comma-separated property declarations.
+                //       nikic/php-parser does this with a single Property statement
+                //       that is capable of holding multiple property declarations.
+                self.semi()?;
+
```

Comment:
this is pretty easy to do actually, and already planned it out, will do it in next PR :)